wpilib/src/robot.rs

```diff
@@ -0,0 +1,155 @@
+// Copyright 2019 First Rust Competition Developers.
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+use super::{
+    ds::{DriverStation, RobotState},
+    notifier::Alarm,
+    RobotBase,
+};
+use std::time;
+use wpilib_sys::{
+    HAL_ObserveUserProgramAutonomous, HAL_ObserveUserProgramDisabled,
+    HAL_ObserveUserProgramStarting, HAL_ObserveUserProgramTeleop, HAL_ObserveUserProgramTest,
+};
+
+/// Implements a specific type of robot program framework, for
+/// `start_iterative` and `start_timed`.
+///
+/// The init methods are called whenever the appropriate mode is entered.
+///
+/// The periodic functions are called for the appropriate mode on an interval.
+pub trait IterativeRobot {
+    fn disabled_init(&mut self) {
+        println!("Default disabled_init method... Override me!");
+    }
+    fn autonomous_init(&mut self) {
+        println!("Default autonomous_init method... Override me!");
+    }
+    fn teleop_init(&mut self) {
+        println!("Default teleop_init method... Override me!");
+    }
+    fn test_init(&mut self) {
+        println!("Default test_init method... Override me!");
```

Comment:
This is a lot of console spam. Many people don't even use all of the inits.

Reply:
It's not really - these only run on mode transition. WPILib do this too (although they also print once in the default periodic functions, which I haven't copied here since that requires significantly more state tracking).